piptools/scripts/compile.py

```diff
@@ -676,6 +676,15 @@ def cli(
         strategy=newline, filenames=(output_file.name, *src_files)
     )
 
+    if strip_extras is None:
+        strip_extras = False
+        log.warning(
+            "WARNING: --strip-extras is becoming the default. "
```

Comment:
Maybe mention "in the next major release" or "in 8.0.0"?

Reply:
Can do!